src/Elliptic/Systems/Xcts/Events/ObserveAdmIntegrals.hpp

```diff
@@ -0,0 +1,192 @@
+ private:
+  using ReductionData = Parallel::ReductionData<
+      // ADM Linear Momentum (x-component)
+      Parallel::ReductionDatum<double, funcl::Plus<>>,
+      // ADM Linear Momentum (y-component)
+      Parallel::ReductionDatum<double, funcl::Plus<>>,
+      // ADM Linear Momentum (z-component)
+      Parallel::ReductionDatum<double, funcl::Plus<>>>;
+
+ public:
+  /// \cond
+  explicit ObserveAdmIntegrals(CkMigrateMessage* msg) : Event(msg) {}
+  using PUP::able::register_constructor;
+  WRAPPED_PUPable_decl_template(ObserveAdmIntegrals);  // NOLINT
+  /// \endcond
+
+  using options = tmpl::list<>;
+  static constexpr Options::String help =
+      "Observe ADM integrals after the XCTS solve.\n"
+      "\n"
+      "Writes reduction quantities:\n"
+      "- Linear momentum";
+
+  ObserveAdmIntegrals() = default;
+
+  using observed_reduction_data_tags =
+      observers::make_reduction_data_tags<tmpl::list<ReductionData>>;
+
+  using compute_tags_for_observation_box = tmpl::list<>;
+
+  using return_tags = tmpl::list<>;
+
+  using argument_tags = tmpl::list<
+      Xcts::Tags::ConformalFactor<DataVector>,
+      gr::Tags::SpatialMetric<DataVector, 3, Frame::Inertial>,
+      gr::Tags::InverseSpatialMetric<DataVector, 3, Frame::Inertial>,
+      gr::Tags::ExtrinsicCurvature<DataVector, 3, Frame::Inertial>,
+      gr::Tags::TraceExtrinsicCurvature<DataVector>,
+      domain::Tags::InverseJacobian<3, Frame::ElementLogical, Frame::Inertial>,
+      domain::Tags::Mesh<3>, domain::Tags::Element<3>,
+      domain::Tags::Faces<3, domain::Tags::FaceNormal<3>>>;
+
+  template <typename Metavariables, typename ArrayIndex,
+            typename ParallelComponent>
+  void operator()(
+      const Scalar<DataVector>& conformal_factor,
+      const tnsr::ii<DataVector, 3>& spatial_metric,
+      const tnsr::II<DataVector, 3>& inv_spatial_metric,
+      const tnsr::ii<DataVector, 3>& extrinsic_curvature,
+      const Scalar<DataVector>& trace_extrinsic_curvature,
+      const InverseJacobian<DataVector, 3, Frame::ElementLogical,
+                            Frame::Inertial>& inv_jacobian,
+      const Mesh<3>& mesh, const Element<3>& element,
+      const DirectionMap<3, tnsr::i<DataVector, 3>>& faces_map_normal,
+      Parallel::GlobalCache<Metavariables>& cache,
+      const ArrayIndex& array_index, const ParallelComponent* const /*meta*/,
+      const ObservationValue& observation_value) const {
+    tnsr::I<double, 3> linear_momentum;
+    for (int I = 0; I < 3; I++) {
+      linear_momentum.get(I) = 0;
+    }
+    compute_linear_momentum(make_not_null(&linear_momentum), conformal_factor,
+                            spatial_metric, inv_spatial_metric,
+                            extrinsic_curvature, trace_extrinsic_curvature,
+                            inv_jacobian, mesh, element, faces_map_normal);
+
+    // Save components of linear momentum as reduction data
+    ReductionData reduction_data{linear_momentum.get(0), linear_momentum.get(1),
```

Comment:
Use the version `get<0>(linear_momentum)` when you know the index at compile time.